extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/devmode/QuteErrorPageSetup.java

```diff
@@ -0,0 +1,223 @@
+package io.quarkus.qute.runtime.devmode;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.dev.ErrorPageGenerators;
+import io.quarkus.dev.spi.HotReplacementContext;
+import io.quarkus.dev.spi.HotReplacementSetup;
+import io.quarkus.qute.Engine;
+import io.quarkus.qute.Escaper;
+import io.quarkus.qute.EvalContext;
+import io.quarkus.qute.ReflectionValueResolver;
+import io.quarkus.qute.Template;
+import io.quarkus.qute.ValueResolver;
+import io.quarkus.runtime.TemplateHtmlBuilder;
+
+public class QuteErrorPageSetup implements HotReplacementSetup {
+
+    private static final Logger LOG = Logger.getLogger(QuteErrorPageSetup.class);
+
+    private static final String TEMPLATE_EXCEPTION = "io.quarkus.qute.TemplateException";
+    private static final String ORIGIN = "io.quarkus.qute.TemplateNode$Origin";
+
+    private static final String PROBLEM_TEMPLATE = ""
```

Comment:
Can we put this in a template file? I strongly hate that in Quarkus core we have HTML in Java strings, but I understand why, no matter how mind-numbing frustrating this is. But in Qute this feels wrong, when this can be in a standalone Qute template file. Or is there a reason I'm not seeing?

Reply:
I wasn't sure where to put the template contents. We could extract it later...